packages/react-server/core/ReactServerAgent/Request.js

```diff
@@ -206,34 +206,48 @@ function buildSuperagentRequest() {
 		req.agent(this._agent);
 	}
 
-	// superagent has some weird, implicit file upload support
-	// that only works if you don't set `type`.
-	if (this._type && this._type !== 'form-data') {
-		req.type(this._type);
-	}
-
 	req.set(this._headers);
-	this._queryParams.forEach(params => req.query(params));
 
-	var postParams = this._postParams;
+	switch (this._method) {
+		case 'GET':
+		case 'HEAD':
+			this._queryParams.forEach(params => req.query(params));
+			break;
+
+		case 'POST':
+		case 'PATCH':
+		case 'PUT':
+			// superagent has some weird, implicit file upload support
+			// that only works if you don't set `type`.
+			if (this._type && this._type !== 'form-data') {
+				req.type(this._type);
+			}
+
+
+			var postParams = this._postParams;
+
+			// convert params to FormData if the request type is form-data
+			if (this._type === "form-data") {
+				if (!SERVER_SIDE) {
+					var formData = new FormData();
+					if (postParams) {
+						var paramKeys = Object.keys(postParams);
+						paramKeys.forEach(key => {
+							formData.append(key, postParams[key]);
+						});
+					}
+					postParams = formData;
+				} else {
+					throw new Error(`ReactServerAgent.type("form-data") not allowed server-side`);
+				}
+			}
 
-	// convert params to FormData if the request type is form-data
-	if (this._type === "form-data") {
-		if (!SERVER_SIDE) {
-			var formData = new FormData();
 			if (postParams) {
-				var paramKeys = Object.keys(postParams);
-				paramKeys.forEach(key => {
-					formData.append(key, postParams[key]);
-				});
+				req.send(postParams);
 			}
-			postParams = formData;
-		} else {
-			throw new Error(`ReactServerAgent.type("form-data") not allowed server-side`);
-		}
-	}
 
-	req.send(postParams);
+			break;
```

Comment:
Fair point.  We don't have functions for those methods, so I didn't implement a check for those.  I can make this more generic.